contractcourt/chain_arbitrator.go

```diff
@@ -342,23 +342,6 @@ func (c *ChainArbitrator) Start() error {
 				pCache:    c.cfg.PreimageDB,
 				signer:    c.cfg.Signer,
 				isOurAddr: c.cfg.IsOurAddress,
-				notifyChanClosed: func() error {
```

Comment:
What's the rationale for this commit? What will clean up the set of watcher, and also stop the existing watcher?

Reply:
The state is cleaned up within `ChainArbitrator.resolveContract`, which is called from `ChannelArbitrator.cfg.MarkChannelResolved`.

Reply:
Is this commit needed for this PR?

I agree that it looks like a nice cleanup, but since it is not trivial and (looks) unrelated to the rest of the commits, I think it would be better in its own PR frankly.

Reply:
Scratch that, I see that `MarkChanFullyClosed` now will prune link nodes, which makes this change related very much 🤣